x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolverTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.analysis.index;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.type.EsField;
+import org.elasticsearch.xpack.sql.type.TypesTests;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class IndexResolverTests extends ESTestCase {
+
+    @Test
+    public void testMergeSameMapping() throws Exception {
+        Map<String, EsField> oneMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        Map<String, EsField> sameMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        assertNotSame(oneMapping, sameMapping);
+        assertEquals(oneMapping, sameMapping);
+
+        String wildcard = "*";
+        IndexResolution resolution = IndexResolver.merge(
+                Arrays.asList(IndexResolution.valid(new EsIndex("a", oneMapping)), IndexResolution.valid(new EsIndex("b", sameMapping))),
+                wildcard);
+
+        assertTrue(resolution.isValid());
+
+        EsIndex esIndex = resolution.get();
+
+        assertEquals(wildcard, esIndex.name());
+        assertEquals(sameMapping, esIndex.mapping());
+    }
+
+    @Test
+    public void testMergeDifferentMapping() throws Exception {
+        Map<String, EsField> oneMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        Map<String, EsField> sameMapping = TypesTests.loadMapping("mapping-basic.json", true);
```

Comment:
Can you add one more test for two mappings that are logically the same (same field types, same field names) but the `json`s themselves are different (different fields order for example)?

Reply:
ES mapping are always returned in order (which is important since the underlying `Map`s depend on it). Hence why the `loadMapping` has the `true` parameter otherwise the order is intentionally randomized (which causes the map equality to fail).